includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -138,24 +138,18 @@ private function process_node( $node ) {
 				// This should not happen very often, but...
 				// If we're here, then we're not sure which spec should
 				// be used. Let's use the top scoring ones.
-				foreach( $spec_ids_sorted as $id ) {
-					$attr_spec_list = array_merge( $attr_spec_list, $rule_spec_list_to_validate[ $id ][AMP_Rule_Spec::ATTR_SPEC_LIST] );
+				foreach ( $spec_ids_sorted as $id ) {
+					$spec_list = isset( $rule_spec_list_to_validate[ $id ][ AMP_Rule_Spec::ATTR_SPEC_LIST ] ) ? $rule_spec_list_to_validate[ $id ][ AMP_Rule_Spec::ATTR_SPEC_LIST ] : null;
+					if ( ! $this->is_missing_mandatory_attribute( $spec_list, $node ) ) {
```

Comment:
This addresses an edge case where there are more than one spec lists, and one of them requires a value. If one of them doesn't require a value, use that instead. 

This detects that with `$this->is_missing_mandatory_attribute()`. If the node if it satisfies one of the specs, this doesn't remove it.